src/osmium/replication/server.py

```diff
@@ -282,12 +282,12 @@ def get_state_info(self, seq=None, retries=2):
             returns `None`. `retries` sets the number of times the download
             is retried when pyosmium detects a truncated state file.
         """
-        for _ in range(retries + 1):
+        for trial in range(retries + 1):
             try:
                 response = self.open_url(self.make_request(self.get_state_url(seq)))
             except Exception as err:
                 LOG.debug("Loading state info %s failed with: %s", seq, str(err))
-                return None
+                continue # retry again in case it is a temporary network issue
```

Comment:
Hmm, I wonder if it would be possible here to distinguish between transient and permanent error. What exception did you see in your case?

Reply:
The error I'm getting is the following one: `<urlopen error [Errno 110] Connection timed out>`

I was thinking of limiting to urllib.error.URLError, but we may also want to retry when getting a HTTP error 500.